frontend/src/components/Cart.tsx

```diff
@@ -0,0 +1,293 @@
+import { useCart } from '../context/CartContext';
+import { useTheme } from '../context/ThemeContext';
+import { useNavigate } from 'react-router-dom';
+
+export default function Cart() {
+  const { cart, loading, updateQuantity, removeItem, clearCart, getTotalPrice } = useCart();
+  const { darkMode } = useTheme();
+  const navigate = useNavigate();
+
+  const handleQuantityChange = async (cartItemId: number, currentQuantity: number, change: number) => {
+    const newQuantity = currentQuantity + change;
+    if (newQuantity > 0) {
+      try {
+        await updateQuantity(cartItemId, newQuantity);
+      } catch (error) {
+        console.error('Failed to update quantity:', error);
+      }
+    }
+  };
+
+  const handleRemoveItem = async (cartItemId: number) => {
+    try {
+      await removeItem(cartItemId);
+    } catch (error) {
+      console.error('Failed to remove item:', error);
+    }
+  };
+
+  const handleClearCart = async () => {
+    if (window.confirm('Are you sure you want to clear your cart?')) {
+      try {
+        await clearCart();
+      } catch (error) {
+        console.error('Failed to clear cart:', error);
+      }
+    }
+  };
+
+  if (loading) {
+    return (
+      <div
+        className={`min-h-screen ${darkMode ? 'bg-dark' : 'bg-gray-100'} pt-20 px-4 transition-colors duration-300`}
+      >
+        <div className="max-w-7xl mx-auto">
+          <div className="flex justify-center items-center h-64">
+            <div className="animate-spin rounded-full h-32 w-32 border-t-2 border-b-2 border-primary"></div>
```

Comment:
The loading spinner lacks accessibility attributes. Screen reader users won't know that content is loading. Add a live region announcement for better accessibility:
```tsx
<div className="flex justify-center items-center h-64">
  <div 
    className="animate-spin rounded-full h-32 w-32 border-t-2 border-b-2 border-primary"
    role="status"
    aria-label="Loading cart"
  >
    <span className="sr-only">Loading cart...</span>
  </div>
</div>
```
```suggestion
          <div className="flex justify-center items-center h-64" aria-live="polite" role="status">
            <div className="animate-spin rounded-full h-32 w-32 border-t-2 border-b-2 border-primary" aria-label="Loading cart"></div>
            <span className="sr-only">Loading cart...</span>
```